### ISSUE TITLE
Data access and other logic in controller

### ISSUE THREAD

**Issue by justinkhart** [justinkhart/OverwatchChumps#4]
So HomeController is a bit fat. Needs to go on a diet. Move complex logic somewhere else.

**Comment by studro** [comment]
Index() seems to the be the only method that is large (but still doesn't seem overly large). Is this the only one you have a problem with? I'm guessing you'd like the database aspects of it (so that the controller knows nothing about the db) placed into a service of some sort - correct me if I'm wrong on this.

**Comment by justinkhart** [comment]
Exactly, we don't want controllers accessing the db at all. A service layer would be great, even better if it uses the repository pattern beneath it. 

Views <- Controllers -> Service layer -> Repository layer -> EntityFramework -> Azure SQL
